pkg/cloud/services/ec2/instances_test.go

```diff
@@ -5320,3 +5320,35 @@ func mockedGetPrivateDNSDomainNameFromDHCPOptionsEmptyCalls(m *mocks.MockEC2APIM
 		},
 	}, nil)
 }
+
+func TestGetCapacityReservationSpecification(t *testing.T) {
+	mockCapacityReservationID := "cr-123"
+	testCases := []struct {
+		name                  string
+		capacityReservationID string
+		expectedRequest       *ec2.CapacityReservationSpecification
+	}{
+		{
+			name:                  "with no CapacityReservationID options specified",
+			capacityReservationID: "",
+			expectedRequest:       nil,
+		},
+		{
+			name:                  "with an valid CapacityReservationID specified",
+			capacityReservationID: *aws.String(mockCapacityReservationID),
```

Comment:
```suggestion
			capacityReservationID: mockCapacityReservationID,
```

Reply:
capacityReservationID accepts only the *string. So it will be  `aws.String(mockCapacityReservationID)`